File: result/translate/chatgpt/java2cpp/c++/MIRROR_CHARACTERS_STRING.cpp
std::string f_gold(std::string str, int n) {
    std::string reverseAlphabet = "zyxwvutsrqponmlkjihgfedcba";
    int l = str.length();
    std::string answer = "";
    for (int i = 0; i < n; i++) {
        if (i < l) answer = answer + str[i];
    }
    for (int i = n; i < l; i++) {
        answer = answer + reverseAlphabet[str[i] - 'a'];
    }
    return answer;
}

File: dataset/java2cpp/java/MIRROR_CHARACTERS_STRING.java
String f_gold ( String str , int n ) {
  String reverseAlphabet = "zyxwvutsrqponmlkjihgfedcba" ;
  int l = str . length ( ) ;
  String answer = "" ;
  for ( int i = 0 ; i < n ; i ++ ) {
    if ( i < l ) answer = answer + str . charAt ( i ) ;
  }
  for ( int i = n ;
  i < l ;
  i ++ ) answer = answer + reverseAlphabet . charAt ( str . charAt ( i ) - 'a' ) ;
  return answer ;
}